src/methods/patch-application.js

```diff
@@ -0,0 +1,20 @@
+export default (self, _id, body) => {
+  const method = 'patch'
+  const promises = []
+  const { ecomAuth } = self
+  const { data, hidden_data } = body
+
+  if (data && typeof data === 'object') {
+    promises.push(ecomAuth.apiRequest(`/applications/${_id}/data.json`, method, data))
+  }
+
+  if (hidden_data && typeof hidden_data === 'object') {
+    promises.push(ecomAuth.apiRequest(`/applications/${_id}/hidden_data.json`, method, hidden_data))
+  }
+
+  if (!data && !hidden_data && body) {
```

Comment:
se alterar data ou hidden data não pode alterar o resto? foi intencional?
acho que o correto seria:

```suggestion
  if (Object.keys(body).find(prop => prop !== 'data' && prop !== 'hidden_data')) {
```

ou algo do tipo...

Reply:
ou você pode no início usar um `for in` no body, salvar data e hidden data outro objeto e deletar do body, sacou?
aí neste ponto você só verificaria se o `body` não ficou vazio, e já teria ele pronto pra requisição (já sem data e hidden data)